File: 백준/Bronze/1157. 단어 공부/단어 공부.py
import sys
from collections import defaultdict

# 입력 받기
word = sys.stdin.readline().strip().upper()

# 문자 개수를 저장할 딕셔너리
char_count = defaultdict(int)

# 문자 개수 세기
for char in word:
    char_count[char] += 1

# 최빈 문자 찾기
max_count = 0
max_char = ''
is_multi_max_char = False

for char, count in char_count.items():
    if count > max_count:
        max_count = count
        max_char = char
        is_multi_max_char = False  # 새로운 최빈 문자가 나왔으므로 초기화
    elif count == max_count:
        is_multi_max_char = True  # 공동 최빈 문자 존재

# 결과 출력
print("?") if is_multi_max_char else print(max_char)

File: 백준/Bronze/1157. 단어 공부/단어 공부.swift
import Foundation

guard let wordInput = readLine() else { exit(0) }

let word = wordInput.uppercased()
var dict: [Character: Int] = [:]
var maxChar: Character = " "
var maxCharCount = 0
var isMultiMaxChar = false

for char in word {
    dict[char, default: 0] += 1
}

for (key, value) in dict {
    if value > maxCharCount {
        maxCharCount = value
        maxChar = key
        isMultiMaxChar = false
    } else if value == maxCharCount {
        isMultiMaxChar = true
    }
}

// 공동 최빈 문자가 있다면 ? 출력, 아니라면 최빈 문자 출력
isMultiMaxChar ? print("?") : print(maxChar)